### PR TITLE
gabbi-run support for multiple input files (take two)

### PR DESCRIPTION
**Pull request by FND**
this addresses #104, improving upon #179 (cf. https://github.com/cdent/gabbi/issues/104#issuecomment-251101913)

no documentation yet because I wasn't sure how or where

### CODE REVIEW

**Comment by cdent** [comment]
docs go in docs/source/runner.rst you'll have to experiment a bit I think to see how you want to express the options for how files work

File: gabbi/runner.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 # WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 # License for the specific language governing permissions and limitations
 # under the License.
-"""Implementation of a command-line runner of single gabbi files."""
+"""Implementation of a command-line runner of single Gabbi files."""
```
Comment (cdent) [inline gabbi/runner.py]: you've introduced a typo here

File: gabbi/runner.py [hunk 2]
```diff
@@ -59,6 +59,13 @@ def run():
 
     Output is formatted as unittest summary information.
     """
+    argv = sys.argv
+    try:  # extract file paths, separated by `--`
+        i = argv.index("--")
+        input_files = argv[i + 1:]
+        argv = argv[:i]
+    except ValueError:  # no file paths
+        input_files = [sys.stdin]
```
Comment (cdent) [inline gabbi/runner.py]: I'd prefer you do this after the argparse argument declarations (or extract those to another method, call that and then argv munging).

File: gabbi/runner.py [hunk 3]
```diff
@@ -89,26 +96,48 @@ def run():
...
 
-    # Initialize response handlers.
-    handler_objects = initialize_handlers(args.response_handlers)
+    response_handlers = initialize_handlers(args.response_handlers)
 
-    data = utils.load_yaml(handle=sys.stdin)
+    failfast = args.failfast
+    failure = False
+    for input_file in input_files:
+        params = (response_handlers, host, port, prefix, force_ssl, failfast)
```
Comment (cdent) [inline gabbi/runner.py]: still no super keen on this param globbing, when execute is called you have to look elsewhere to see what it is called with, that seems bad to me.

File: gabbi/runner.py [hunk 4]
```diff
@@ -89,26 +96,48 @@ def run():
...
+
+        if not failure:  # once failed, this is considered immutable
+            failure = not success
+        if failure and failfast:
+            break
+
+    sys.exit(failure)
+
+
+def execute(handle, response_handlers, host, port, prefix,  # TODO(FND): rename
```
Comment (cdent) [inline gabbi/runner.py]: either accept the name or change it, do not leave for later. If you're trying to disambiguate, try `execute_tests` or `execute_test_file` or things in that vein

File: gabbi/runner.py [hunk 5]
```diff
@@ -89,26 +96,48 @@ def run():
...
         if 'defaults' in data:
             data['defaults']['ssl'] = True
         else:
             data['defaults'] = {'ssl': True}
+
     loader = unittest.defaultTestLoader
     test_suite = suitemaker.test_suite_from_dict(
         loader, 'input', data, '.', host, port, None, None, prefix=prefix,
-        handlers=handler_objects)
+        handlers=response_handlers)
```
Comment (cdent) [inline gabbi/runner.py]: doing `s/handler_objects/response_handlers/` on the file is an unrelated change that you seem to only be doing because you don't like/understand the name. Why is the change a part of this patchset?

Comment (FND) [inline gabbi/runner.py]: You're right, this doesn't belong here - will fix.

File: gabbi/runner.py (hunk 4 above)
Comment (FND) [inline gabbi/runner.py]: As agreed: `run_suite`

**Comment by FND** [comment]
Thanks for the review. I've tried to address your concerns, rewriting history for minor changes (`run_suite`, `handler_objects`) and adding new commits for the rest.

As for docs, I'm not gonna do that tonight - so I'd prefer to create a separate issue for that, to be addressed another day.

File: gabbi/runner.py (hunk 1 above)
Comment (FND) [inline gabbi/runner.py]: Assuming you're referring to "single" being obsolete, does that really count as a typo?

Comment (cdent) [inline gabbi/runner.py]: You appear to accidentally capitalised gabbi where it wasn't before. I assume this was a mistake of some kind?

File: gabbi/runner.py [hunk 6]
```diff
@@ -89,26 +92,48 @@ def run():
...
+    args = parser.parse_args(argv[1:])
     host, port, prefix, force_ssl = utils.host_info_from_target(
         args.target, args.prefix)
 
-    # Initialize response handlers.
     handler_objects = initialize_handlers(args.response_handlers)
 
-    data = utils.load_yaml(handle=sys.stdin)
+    _run = lambda fh: run_suite(fh, handler_objects, host, port, prefix,
+                                force_ssl, failfast)
```
Comment (cdent) [inline gabbi/runner.py]: see my comment elsewhere about obscurity 👎
